### ISSUE TITLE
How `partition_avx512` is auto-tuned?

### ISSUE THREAD

**Issue by condy0919** [pytorch/FBGEMM#2101]
I am currently learning to port the fp16 multiplication of fbgemm to riscv. I found that the fp16 gemm `uses cblas_gemm_compute`, which uses `partition_avx512` to partition the `mb_max=120` rows of matrix A into several tiles. e.g., 49 rows is mapping to 3 x 13rows + 1 x 10rows. It could be 4 x 12rows + 1 x 1row? From my perspective, these data seem to be irregular.

**Comment by jspark1105** [comment]
This is based on tuning results on x86 CPUs so you can change to whichever better for the riscv processor you're optimizing for.

**Comment by jianyuh** [comment]
Yes: previously we tune something like https://github.com/pytorch/FBGEMM/pull/82/files (this is for avx2). You can adjust for your customized HW.

**Comment by condy0919** [comment]
> Basically, this Diff switches the register layout in C accumulation buffer inside micro-kernel from MR * 1 to MR * 2. Check the reasons in T40816746.

Could you provide the reasons in `T40816746`?
